src/blur/meta-blur-actor.c

```diff
@@ -0,0 +1,933 @@
+//
+//  Copyright (C) 2015 Deepin Technology Co., Ltd.
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+#include <config.h>
+
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+#include <GL/gl.h>
+#include <clutter/clutter.h>
+
+#include "cogl-utils.h"
+#include "blur-utils.h"
+#include <meta/errors.h>
+#include <meta/screen.h>
+#include <meta/compositor-mutter.h>
+#include <meta-blur-actor.h>
+
+static void _stage_add_always_redraw_actor(ClutterStage *stage, ClutterActor *actor)
```

Comment:
The references to normally not available API, functionality in Clutter. Those will be called only if those methods are present in the clutter package, see patch in the PR description.